client/client.go

```diff
@@ -166,20 +182,21 @@ func GetAPIClient(apiURL string, apiKey string, apiVersion string) (*APIClient,
 	return getAPIClient(apiURL, apiKey, apiVersion, nil)
 }
 
-func getAPIClient(apiURL string, apiKey string, apiVersion string, tlsConfig *tls.Config) (*APIClient, error) {
+func getAPIClient(apiURL string, apiKey string, apiVersion string, tr http.RoundTripper) (*APIClient, error) {
 	versionedURL := gomaasapi.AddAPIVersionToURL(apiURL, apiVersion)
 
 	authClient, err := gomaasapi.NewAuthenticatedClient(versionedURL, apiKey)
 	if err != nil {
 		return nil, err
 	}
 
-	if tlsConfig != nil {
-		tr := &http.Transport{TLSClientConfig: tlsConfig}
-		httpClient := &http.Client{Transport: tr}
-		authClient.HTTPClient = httpClient
+	if tr == nil {
+		tr = &http.Transport{}
```

Comment:
I think we should use https://pkg.go.dev/net/http#DefaultTransport here, because `&http.Transport{}` will ignore system environment variables for HTTP proxy (e.g. HTTP_PROXY, HTTPS_PROXY and NO_PROXY)

Reply:
Sounds good, I wasn't aware of that